backend/app/services/chat/url_parser.py

```
@@ -0,0 +1,429 @@
+            for script in soup(["script", "style", "nav", "footer", "header"]):
+                script.decompose()
+
+            # Get text
+            text = soup.get_text(separator="\n")
+
+            # Clean up whitespace
+            lines = (line.strip() for line in text.splitlines())
+            chunks = (phrase.strip() for line in lines for phrase in line.split("  "))
+            text = "\n".join(chunk for chunk in chunks if chunk)
+
+            return text
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)